rest/specs/src/main/filtered-resources/openapi/openapi.yaml

```diff
@@ -0,0 +1,250 @@
+    BadRequest:
+      description: >
+        The client message could not understood by the server due to invalid format or syntax.
+      headers:
+        'Content-Version':
+          $ref: '#/components/headers/ContentVersion'
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/ErrorMessage'
+    DefaultError:
+      description: >
+        Any other possible errors not currently known.
+      headers:
+        'Content-Version':
+          $ref: '#/components/headers/ContentVersion'
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/ErrorMessage'
+  parameters:
+    Accept-Version:
+      name: Accept-Version
+      in: header
+      description: >
+        The minimal API version that a client using this API will accept
+        responses from.
+      required: true
+      schema:
+        $ref: '#/components/schemas/Version'
+  headers:
+    ContentVersion:
+      description: >
+        The API version used by the server to produce the REST message. The server will accept
+        messages for any minor versions prior to this one.
+      required: true
+      schema:
+        $ref: '#/components/schemas/Version'
+      examples:
+        major.minor:
+          value: '1.2'
+          summary: A released minor version
+        major.minor.patch:
+          value: '1.0.3'
+          summary: A patched version
+        dev:
+          value: '1.2.0-SNAPSHOT'
+          summary: A development version
+  schemas:
+    Product:
+      description: The JSON description of the Product being ingested.
+      type: object
+      properties:
+        productInfo:
+          $ref: '#/components/schemas/ProductInfo'
+        file:
+          $ref: '#/components/schemas/File'
+      required:
+        - productInfo
+        - file
+    ProductInfo:
+        properties:
+          title:
+            type: string
+            minLength: 1
+            maxLength: 255
+            description: The title of the Product.
+            example: A Picture
+          filename:
+            type: string
+            minLength: 1
+            maxLength: 255
+            description: The name of the source file.
+            example: product.jpeg
+          fileSize:
+            type: integer
+            description: The size of the file payload in bytes.
+            format: int32
+            minimum: 1
+            example: 10000000000
+          mimeType:
+            type: string
+            pattern: '^(?=[-a-z]{1,127}/[-\.a-z0-9]{1,127}$)[a-z]+(-[a-z]+)*/[a-z0-9]+([-\.][a-z0-9]+)*$'
+            description: The MIME type of the file being sent.
+            example: image/jpeg
+        required:
+          - filename
+          - bytes
+          - mimeType
+    File:
+      description: The actual file attachment that is sent in the request.
+      type: string
+      format: binary
+      minLength: 1
+      maxLength: 10000000000
```

Comment:
Since each character in a base64 string represents 6 bits. 10 GB would be represented using 10 * 1024 * 1024 * 1024 * 8 / 6 = 14,316,557,653.3333 characters + 1 if padding is used. Please add to the description to indicate the current max is set to 10GB which equals the specified number of characters.